timemachine/md/moves.py

```diff
@@ -12,69 +14,96 @@
 
 from timemachine import lib
 from timemachine.lib import custom_ops
-from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.md.states import CoordsVelBox
-from timemachine.potentials import BoundPotential, HarmonicBond
+from timemachine.potentials import BoundPotential
 
 _State = TypeVar("_State")
 
 
-class MonteCarloMove(Generic[_State]):
-    n_proposed: int = 0
-    n_accepted: int = 0
+class Move(Generic[_State], ABC):
+    @abstractmethod
+    def move(self, _: _State) -> _State:
+        ...
+
+    def sample_chain_iter(self, x: _State) -> Iterator[_State]:
+        while True:
+            x = self.move(x)
+            yield x
+
+    def sample_chain(self, x: _State, n_samples: int) -> List[_State]:
+        return list(islice(self.sample_chain_iter(x), n_samples))
+
 
+@dataclass
```

Comment:
What does dataclass get us in this case? This seems less like a class that contains data than a collection of functionality.

Reply:
Not much in retrospect. I wanted to be able to annotate the types of `n_proposed` and `n_accepted`, but it appears that's possible to do without making it a dataclass. Will revert.

Reply:
Before you revert it, it looks like it does make `help(MonteCarloMove)` much cleaner, so it might be worth it?  Find it a bit of an odd use of the `dataclass` abstraction, but looks like it might make the use/documentation of it cleaner.

Reply:
Hmm, they seem [about the same](https://gist.github.com/mcwitt/2c47ff750cd19bc1c2b4a0435f5666f3) (i.e. not that clean) to me? With the plain class, help includes "Methods inherited from Move", which seems to be missing with the dataclass.